src/main/java/com/impactupgrade/nucleus/service/segment/XeroAccountingPlatformService.java

```diff
@@ -493,12 +442,37 @@ protected List<LineItem> getLineItems(AccountingTransaction accountingTransactio
         if (accountingTransaction.transactionType == EnvironmentConfig.TransactionType.TICKET) {
             lineItem.setAccountCode("160");
             lineItem.setItemCode("EI");
+
+            if ("true".equalsIgnoreCase(getCustomDonationField(accountingTransaction, OTHER_INCOME_FIELD_NAME))) {
+                lineItem.setAccountCode("260");
+                lineItem.setItemCode("Other Income");
+            }
+
         } else if (accountingTransaction.recurring) {
             lineItem.setAccountCode("122");
             lineItem.setItemCode("Partner");
+
+            if ("true".equalsIgnoreCase(getCustomDonationField(accountingTransaction, WPG_FIELD_NAME))) {
+                lineItem.setAccountCode("120");
+                lineItem.setItemCode("RecurringWPG");
+            }
```

Comment:
Just CC'd you on a question to Maria on https://docs.google.com/document/d/1wEjPtMvVwZr-k3lFa5D6M9NgP6IGZn_v/edit. This code would be creating a single invoice coded to 120 if a recurring gift had WPG=true in SFDC. Which MIGHT be correct. But as I reread what they had in their mappings, wondering if what they want is BOTH an invoice coded to 122 AND a "receive money" coded to 120.

IIRC, they want the latter, so we'd need to break this up to create both an invoice and a receive money with their independent line items. That's currently done by dr-hub's overridden `createTransaction`, but we could get rid of that and just do it here...